internal/cmd/request.go

```diff
@@ -24,11 +24,20 @@ func MustHandleCMDRequestAndExit(endpoint string, request any, responseType any)
 }
 
 func MustHandleCMDRequestAndExitFull(endpoint string, request any, responseType any, options CommonOptions, customPrintFunc CustomResponseFormatter) {
+	if !IsTesting() {
+		MustStartCMDServer()
+	}
+
 	response, err := SendCMDRequest(endpoint, request)
 	if err != nil {
+		MustStopCMDServer()
```

Comment:
What happens if we call `MustStopCMDServer()` when we are not in testing mode (i.e. if this function did not call `MustStartCMDServer()`?

Reply:
I'm not sure I understand. If we aren't in testing mode the only other way we'd get here is by running the cmd with go run (which would in turn call `MustStartCMDServer()`). I don't see anywhere else that would use these cmd start/stop servers, but even if there were, stopping the server is designed to be able to be called as many times as we want without causing an error/panic. You can see something similar being done [here](https://github.com/edulinq/autograder-server/blob/main/internal/cmd/test.go#L36) where we stop the server before starting it just in case the previous test wasn't able to stop it for some reason.

Reply:
kk